ext/standard/string.c

```diff
@@ -2353,12 +2353,22 @@ PHP_FUNCTION(substr_replace)
 		zend_string *tmp_repl_str = NULL;
 		if (repl_ht) {
 			repl_idx = 0;
-			while (repl_idx < repl_ht->nNumUsed) {
-				tmp_repl = &repl_ht->arData[repl_idx].val;
-				if (Z_TYPE_P(tmp_repl) != IS_UNDEF) {
-					break;
+			if (HT_IS_PACKED(repl_ht)) {
```

Comment:
This seems to be a recurring pattern. We could add a macro for it (or maybe it should use zend_hash_get_current_data/zend_hash_move_forward?)